libdevice/sanitizer_utils.cpp

```diff
@@ -6,22 +6,627 @@
+      const char __SYCL_CONSTANT__ *func) {                                    \
+    u##size *shadow_address = (u##size *)MemToShadow(addr, as);                \
+    if (shadow_address && *shadow_address) {                                   \
+      __asan_report_access_error(addr, as, size, is_write, addr, file, line,   \
+                                 func, true);                                  \
+    }                                                                          \
+  }
+
+ASAN_REPORT_ERROR_BYTE(load, false, 8)
+ASAN_REPORT_ERROR_BYTE(load, false, 16)
+ASAN_REPORT_ERROR_BYTE(store, true, 8)
+ASAN_REPORT_ERROR_BYTE(store, true, 16)
+
+#define ASAN_REPORT_ERROR_N(type, is_write)                                    \
+  DEVICE_EXTERN_C_NOINLINE void __asan_##type##N(                              \
+      uptr addr, size_t size, int32_t as, const char __SYCL_CONSTANT__ *file,  \
+      int32_t line, const char __SYCL_CONSTANT__ *func) {                      \
+    if (auto poisoned_addr = __asan_region_is_poisoned(addr, as, size)) {      \
+      __asan_report_access_error(addr, as, size, is_write, poisoned_addr,      \
+                                 file, line, func);                            \
+    }                                                                          \
+  }                                                                            \
+  DEVICE_EXTERN_C_NOINLINE void __asan_##type##N_noabort(                      \
+      uptr addr, size_t size, int32_t as, const char __SYCL_CONSTANT__ *file,  \
+      int32_t line, const char __SYCL_CONSTANT__ *func) {                      \
+    if (auto poisoned_addr = __asan_region_is_poisoned(addr, as, size)) {      \
+      __asan_report_access_error(addr, as, size, is_write, poisoned_addr,      \
+                                 file, line, func, true);                      \
+    }                                                                          \
+  }
+
+ASAN_REPORT_ERROR_N(load, false)
+ASAN_REPORT_ERROR_N(store, true)
+
+static const __SYCL_CONSTANT__ char __set_shadow_local0[] =
+    "LOCAL: __asan_set_shadow_local_memory(%p, %d, %d)\n";
+static const __SYCL_CONSTANT__ char __set_shadow_local1[] =
+    "LOCAL: memset(%p - %p, 0)\n";
+static const __SYCL_CONSTANT__ char __set_shadow_local2[] =
+    "LOCAL: memset(%p, %d)\n";
+
+DEVICE_EXTERN_C_NOINLINE void
+__asan_set_shadow_local_memory(uptr ptr, size_t size,
+                               size_t size_with_redzone) {
+  // __spirv_ocl_printf(__set_shadow_local0, ptr, size, size_with_redzone);
+
+  uptr aligned_size = RoundUpTo(size, ASAN_SHADOW_GRANULARITY);
+
+  auto user_end = ptr + size - 1;
+  auto *shadow_beg = (s8 *)MemToShadow(ptr, AS_LOCAL);
+  auto *shadow_end = (s8 *)MemToShadow(user_end, AS_LOCAL);
+  // __spirv_ocl_printf(__set_shadow_local1, shadow_beg, shadow_end);
+
+  while (shadow_beg <= shadow_end) {
+    *shadow_beg = 0;
+    ++shadow_beg;
```

Comment:
why isn't this shadow memory initialized using enqueue write?

Reply:
This code is redundant. I'll remove it.